### DISCUSSION TITLE
KTC Stats

### DISCUSSION THREAD

**Discussion by rqtqp** [TypQxQ/KTC#46] [Configuring]
Hi,
Is it possible to add additional configurable values to KTC stats in addition to selects/deselects?

I use klipper+KTC for a large drawing plotter and when it draws large pieces I have to keep an eye on the marker, so I wrote a couple of macroses to keep an eye on the distance the marker travels on the sheet in Z0 state. Since I use markers of the certain manufacturer, calibrate them and I know the distance it can draw without quality loss. Macros puts the print on pause after reaching a certain distance traveled, I change the marker and turn it back on. This suited me before, when I used just 1 or 2 colors, but now I started doing multicolor works, and since the different colors of markers are spent unevenly, it becomes a light problem to maintain a pile of markers) 
But here studying your wonderful extension more in detail, I saw that KTC has its own usage statistics, so I wondered if it is possible to record/store some data on each marker (per tool basis) and reset it when it is replaced.

```
KTC Total Statistics:
599 selects completed(158.5%) in 0s. Avg. 0s.
158 deselects completed(50.0%) in 0s. Avg. 0s.
========================================================
Tool 0:

246 selects completed(168.5%) in 0s. Avg. 0s.
50 deselects completed(50.0%) in 0s. Avg. 0s.
--------------------------------------------------------
Tool 1:

129 selects completed(161.2%) in 0s. Avg. 0s.
40 deselects completed(50.0%) in 0s. Avg. 0s.
--------------------------------------------------------
Tool 2:

217 selects completed(148.6%) in 0s. Avg. 0s.
66 deselects completed(50.0%) in 0s. Avg. 0s.
--------------------------------------------------------
Tool 3:

7 selects completed(116.7%) in 0s. Avg. 0s.
2 deselects completed(50.0%) in 0s. Avg. 0s.
========================================================
```

**Comment by TypQxQ** [comment]
Not out of the box but that would be interesting. Basically just add another parameter and a reset command.
